sympy/vector/tests/test_coordsysrect.py

```diff
@@ -1,462 +1,472 @@
+
+    # Cylindrical with `lambda`
+    a = CoordSys3D('a', lambda r, theta, z:
+        (r*cos(theta), r*sin(theta), z),
+        variable_names=["r", "theta", "z"]
+    )
+    r, theta, z = a.base_scalars()
+    assert a.transformation_to_parent() == Matrix([[r*cos(theta)], [r*sin(theta)], [z]])
+    assert a.lame_coefficients() == (
+        sqrt(sin(theta)**2 + cos(theta)**2),
+        sqrt(r**2*sin(theta)**2 + r**2*cos(theta)**2),
+        1
+    )  # ==> this should simplify to (1, a.x, 1)
+
+    raises(TypeError, lambda: CoordSys3D('a', transformation={
+        x: x*sin(y)*cos(z), y:x*sin(y)*sin(z), z:  x*cos(y)}))
+
+
+def test_check_orthogonality():
+    x, y, z = symbols('x y z')
+    u,v = symbols('u, v')
+    a = CoordSys3D('a', transformation=((x, y, z), (x*sin(y)*cos(z), x*sin(y)*sin(z), x*cos(y))))
+    assert a._check_orthogonality(a._transformation) is True
+    a = CoordSys3D('a', transformation=((x, y, z), (x * cos(y), x * sin(y), z)))
+    assert a._check_orthogonality(a._transformation) is True
+    a = CoordSys3D('a', transformation=((u, v, z), (cosh(u) * cos(v), sinh(u) * sin(v), z)))
+    assert a._check_orthogonality(a._transformation) is True
+
+    raises(ValueError, lambda: CoordSys3D('a', transformation=((x, y, z), (x, x, z))))
+    raises(ValueError, lambda: CoordSys3D('a', transformation=(
+        (x, y, z), (x*sin(y/2)*cos(z), x*sin(y)*sin(z), x*cos(y)))))
+
+
+def test_coordsys3d():
+    with warns_deprecated_sympy():
+        assert CoordSysCartesian("C") == CoordSys3D("C")
+
+
+def test_rotation_trans_equations():
+    a = CoordSys3D('a')
+    from sympy import symbols
+    q0 = symbols('q0')
+    assert a._rotation_trans_equations(a._parent_rotation_matrix, a.base_scalars()) == (a.x, a.y, a.z)
+    assert a._rotation_trans_equations(a._inverse_rotation_matrix(), a.base_scalars()) == (a.x, a.y, a.z)
+    b = a.orient_new_axis('b', 0, -a.k)
+    assert b._rotation_trans_equations(b._parent_rotation_matrix, b.base_scalars()) == (b.x, b.y, b.z)
+    assert b._rotation_trans_equations(b._inverse_rotation_matrix(), b.base_scalars()) == (b.x, b.y, b.z)
+    c = a.orient_new_axis('c', q0, -a.k)
+    assert c._rotation_trans_equations(c._parent_rotation_matrix, c.base_scalars()) == \
+           (-sin(q0) * c.y + cos(q0) * c.x, sin(q0) * c.x + cos(q0) * c.y, c.z)
+    assert c._rotation_trans_equations(c._inverse_rotation_matrix(), c.base_scalars()) == \
+           (sin(q0) * c.y + cos(q0) * c.x, -sin(q0) * c.x + cos(q0) * c.y, c.z)
+
+def test_issue_19725():
+    N = CoordSys3D('N')
+    M = N.create_new("M", transformation="cylindrical")
+    a = symbols("a")
+    O = N.orient_new_axis("O", a, N.k)
+
+    assert M.transformation_to_parent() == Matrix([[M.r*cos(M.theta)], [M.r*sin(M.theta)], [M.z]])
+    assert M.transformation_from_parent() == Matrix([[sqrt(N.x**2 + N.y**2)], [atan2(N.y, N.x)], [N.z]])
+    assert isinstance(O.transformation_to_parent(), Matrix)

```

Comment:
Is this the test which you intent to add?

Reply:
Yes, and I made some changes to other tests so that they asserted the transformation_to/from_parent() results were matrices not tuples

Reply:
> @abaokar-07 You changed the output type from `Tuple` to `Matrix` , so update the doctest example which still expects the output to be `Tuple`.
> 
> File "/home/runner/work/sympy/sympy/sympy/vector/coordsysrect.py", line 1007, in sympy.vector.coordsysrect.CoordSys3D.create_new
> Failed example:
> b.transformation_to_parent()
> Expected:
> (b.r_sin(b.theta)_cos(b.phi), b.r_sin(b.phi)_sin(b.theta), b.r_cos(b.theta)) Got: Matrix([ [b.r_sin(b.theta)_cos(b.phi)], [b.r_sin(b.phi)_sin(b.theta)], [ b.r_cos(b.theta)]])
> 
> and similar examples in docstring.

I'm a bit confused. I modified test_coordsysrect.py and was able to have the test run on my local end. How do I change the doctest directly?

Reply:
The situation is a bit weird. May be it is due to copying the file to somewhere else and then replacing the original file with the edited file. You can try just editing the original file. Though, I never faced such things before, not sure why it is happening.

Reply:
Oh I think I was confused. I edited the test_coordsysrect.py file but not the doctests specifically, which does not in fact pass locally. What file do I access to manually change a doctest?

Reply:
sympy/vector/coordsysrect.py", line 1007, in sympy.vector.coordsysrect.CoordSys3D.create_new

Reply:
Ah I see. I've edited the file and before I commit again I'm running the doctests on my end.